### PR TITLE
PLPO-31

### PR DESCRIPTION
**Pull request by lkulig**

### CODE REVIEW

File: planning-poker-app/gulp/tasks/bower.js [hunk 1]
```diff
@@ -1,11 +1,22 @@
 var gulp = require('gulp');
-var mainBowerFiles = require('main-bower-files');
-var gulpFilter = require('gulp-filter');
-var filter = gulpFilter(['**/*.js', '!**/*.min.js', '!**/index.js']);
+var wiredep = require('wiredep').stream;
```
Comment (activey) [inline planning-poker-app/gulp/tasks/bower.js]: how about just require('wiredep') and do pipe(wiredep.stream())?

File: planning-poker-app/profile_dev.gradle [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import org.apache.tools.ant.taskdefs.condition.Os
```
Comment (activey) [inline planning-poker-app/profile_dev.gradle]: can we have just a single file and variables for 'dev'/'prod'?

File: planning-poker-app/src/main/java/com/futureprocessing/planningpoker/config/WebSecurityConfig.java [hunk 3]
```diff
@@ -12,26 +15,38 @@
...
-            .antMatchers(ROOT_URL).permitAll()
-            .anyRequest().authenticated()
+        http.csrf()
+            .disable()
+            .exceptionHandling()
+            .authenticationEntryPoint(authenticationEntryPoint)
             .and()
             .formLogin()
-            .loginPage(LOGIN_URL)
+            .loginProcessingUrl("/api/authentication").permitAll()
```
Comment (activey) [inline planning-poker-app/src/main/java/com/futureprocessing/planningpoker/config/WebSecurityConfig.java]: consts for those urls? maybe some additional java interface that will gather them?

File: planning-poker-app/src/main/java/com/futureprocessing/planningpoker/config/WebSecurityConfig.java [hunk 4]
```diff
@@ -12,26 +15,38 @@
...
             .permitAll()
             .and()
             .logout()
-            .permitAll();
+            .logoutUrl("/api/logout")
+            .deleteCookies("JSESSIONID")
+            .permitAll()
+            .and()
+                .authorizeRequests()
+                .antMatchers("/api/stupid").authenticated();
```
Comment (activey) [inline planning-poker-app/src/main/java/com/futureprocessing/planningpoker/config/WebSecurityConfig.java]: who do you call stupid? :P

File: planning-poker-app/src/main/java/com/futureprocessing/planningpoker/security/auth/AuthenticationEntryPoint.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Component
+public class AuthenticationEntryPoint implements org.springframework.security.web.AuthenticationEntryPoint {
+
+    @Override
+    public void commence(HttpServletRequest request, HttpServletResponse response, AuthenticationException exception) throws IOException,
+            ServletException {
+        response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Access Denied");
```
Comment (activey) [inline planning-poker-app/src/main/java/com/futureprocessing/planningpoker/security/auth/AuthenticationEntryPoint.java]: const for erro msg?

File: planning-poker-app/src/main/java/com/futureprocessing/planningpoker/security/auth/AuthenticationFailureHandler.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Component
+public class AuthenticationFailureHandler extends SimpleUrlAuthenticationFailureHandler {
+
+    @Override
+    public void onAuthenticationFailure(HttpServletRequest request, HttpServletResponse response,
+                                        AuthenticationException exception) throws IOException, ServletException {
+        response.sendError(SC_UNAUTHORIZED, "Authentication failed");
```
Comment (activey) [inline planning-poker-app/src/main/java/com/futureprocessing/planningpoker/security/auth/AuthenticationFailureHandler.java]: and here

File: planning-poker-app/src/main/webapp/scripts/app/app.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    'ui.router'
+]).config(function ($stateProvider, $locationProvider, $urlRouterProvider) {
+    $urlRouterProvider.when('', '/main');
+    $urlRouterProvider.otherwise('/');
+}).run(function ($state, Principal, $rootScope, Auth) {
+    $rootScope.$on('$stateChangeStart', function (event, toState, toStateParams) {
+        if (!Principal.isAuthenticated()) {
+            if (toState.name !== 'login') {
+                event.preventDefault();
+                $state.go('login');
```
Comment (activey) [inline planning-poker-app/src/main/webapp/scripts/app/app.js]: $state.go(toState.name)? ;)

File: planning-poker-app/src/main/webapp/scripts/app/login/login.controller.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+        $scope.login = function () {
+            $scope.processing = true;
+            Auth.login({
+                username: $scope.credentials.username,
+                password: $scope.credentials.password
+            }, function (response) {
+                $scope.authenticationError = false;
+                Principal.authenticate(response);
+                $state.go('main');
```
Comment (activey) [inline planning-poker-app/src/main/webapp/scripts/app/login/login.controller.js]: some enum for possible states?

File: planning-poker-app/src/main/webapp/scripts/components/auth/provider/authentication.provider.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+angular.module('planningPoker')
+    .factory('AuthenticationProvider', function loginService($http) {
+        return {
+            login: function(credentials) {
+                var data = 'j_username=' + encodeURIComponent(credentials.username) +
```
Comment (activey) [inline planning-poker-app/src/main/webapp/scripts/components/auth/provider/authentication.provider.js]: whoa :)

File: planning-poker-app/gulp/tasks/bower.js (hunk 1 above)
Comment (lkulig) [inline planning-poker-app/gulp/tasks/bower.js]: why not.

File: planning-poker-app/profile_dev.gradle (hunk 2 above)
Comment (lkulig) [inline planning-poker-app/profile_dev.gradle]: I would leave this for now. I will create another task for managing dev/produ profiles in gradle and gulp.

File: planning-poker-app/src/main/java/com/futureprocessing/planningpoker/config/WebSecurityConfig.java [hunk 10]
```diff
@@ -12,26 +15,38 @@
...
     private PlanningPokerAuthenticationProvider authenticationProvider;
 
     @Override
     protected void configure(HttpSecurity http) throws Exception {
-        http
-            .authorizeRequests()
-            .antMatchers(ROOT_URL).permitAll()
-            .anyRequest().authenticated()
+        http.csrf()
+            .disable()
```
Comment (lkulig) [inline planning-poker-app/src/main/java/com/futureprocessing/planningpoker/config/WebSecurityConfig.java]: Disabled for now.

File: planning-poker-app/src/main/java/com/futureprocessing/planningpoker/config/WebSecurityConfig.java (hunk 3 above)
Comment (lkulig) [inline planning-poker-app/src/main/java/com/futureprocessing/planningpoker/config/WebSecurityConfig.java]: yeah why not, const seems better for now.

File: planning-poker-app/src/main/java/com/futureprocessing/planningpoker/config/WebSecurityConfig.java (hunk 4 above)
Comment (lkulig) [inline planning-poker-app/src/main/java/com/futureprocessing/planningpoker/config/WebSecurityConfig.java]: aaaarghhhhh. I forgot to delete this :( it was only for testing purpose

File: planning-poker-app/src/main/java/com/futureprocessing/planningpoker/security/auth/AuthenticationEntryPoint.java (hunk 5 above)
Comment (lkulig) [inline planning-poker-app/src/main/java/com/futureprocessing/planningpoker/security/auth/AuthenticationEntryPoint.java]: kk

File: planning-poker-app/src/main/java/com/futureprocessing/planningpoker/security/auth/AuthenticationFailureHandler.java (hunk 6 above)
Comment (lkulig) [inline planning-poker-app/src/main/java/com/futureprocessing/planningpoker/security/auth/AuthenticationFailureHandler.java]: kk

File: planning-poker-app/src/main/webapp/scripts/app/login/login.controller.js (hunk 8 above)
Comment (lkulig) [inline planning-poker-app/src/main/webapp/scripts/app/login/login.controller.js]: some const in angular will probably be enough

File: planning-poker-app/src/main/webapp/scripts/components/auth/provider/authentication.provider.js (hunk 9 above)
Comment (lkulig) [inline planning-poker-app/src/main/webapp/scripts/components/auth/provider/authentication.provider.js]: a thing to refactor a little bit...

File: planning-poker-app/src/main/webapp/scripts/components/auth/principal.service.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+            isAuthenticated: function () {
+                return isAuthenticated;
+            },
+
+            authenticate: function (newIdentity) {
+                identity = newIdentity;
+                isAuthenticated = identity !== null;
+            }
+        };
+    });
```
Comment (dczyz) [inline planning-poker-app/src/main/webapp/scripts/components/auth/principal.service.js]: Tests? :tongue:

Comment (lkulig) [inline planning-poker-app/src/main/webapp/scripts/components/auth/principal.service.js]: yeah, someday... :D

Comment (dczyz) [inline planning-poker-app/src/main/webapp/scripts/components/auth/principal.service.js]: f2f

**Pull request merged**